paper-server/patches/sources/net/minecraft/world/entity/EntitySelector.java.patch

```diff
@@ -23,31 +23,49 @@
  
      private EntitySelector() {
      }
-@@ -27,18 +_,24 @@
+@@ -27,29 +_,31 @@
      }
  
      public static Predicate<Entity> pushableBy(Entity entity) {
 +        // Paper start - Climbing should not bypass cramming gamerule
-+        return pushable(entity, false);
-+    }
-+
-+    public static Predicate<Entity> pushable(Entity entity, boolean ignoreClimbing) {
-+        // Paper end - Climbing should not bypass cramming gamerule
          Team team = entity.getTeam();
          Team.CollisionRule collisionRule = team == null ? Team.CollisionRule.ALWAYS : team.getCollisionRule();
          return (Predicate<Entity>)(collisionRule == Team.CollisionRule.NEVER
              ? Predicates.alwaysFalse()
              : NO_SPECTATORS.and(
 -                entity1 -> {
 -                    if (!entity1.isPushable()) {
-+                entity1 -> { final Entity pushedEntity = entity1; // Paper - OBFHELPER
-+                    if (!pushedEntity.isCollidable(ignoreClimbing) || !pushedEntity.canCollideWithBukkit(entity) || !entity.canCollideWithBukkit(pushedEntity)) { // CraftBukkit - collidable API // Paper - Climbing should not bypass cramming gamerule
++            entity1 -> { final Entity pushedEntity = entity1; // Paper - OBFHELPER
++                if (!pushedEntity.isPushable() || !pushedEntity.canCollideWithBukkit(entity) || !entity.canCollideWithBukkit(pushedEntity)) { // CraftBukkit - collidable API // Paper - Climbing should not bypass cramming gamerule
```

Comment:
Comment is not necessary, this line is not touched for the climbing cramming bypass option (as `pushedEntity#isPushable` is the Vanilla code and the other change is just the collidable API)